lib/market.rb

```diff
@@ -1,3 +1,29 @@
 class Market
-  
+  attr_reader :name, :vendors
+
+  def initialize(name)
+    @name = name
+    @vendors = []
+  end
+
+  def add_vendor(vendor)
+    @vendors.push(vendor)
+  end
+
+  def vendor_names
+    names = []
+    @vendors.each do |vendor|
+      names.push(vendor.name)
+    end
+    names
+  end
+
+  def vendors_that_sell(item)
+    vendors_with_items = []
+    @vendors.select do |vendor|
+      vendor.inventory[item].zero?
+      vendors_with_items.push(vendor)
+      # require 'pry'; binding.pry
+    end
```

Comment:
```suggestion
    @vendors.select do |vendor|
      !vendor.inventory[item].zero?
    end
```

This method was quite close!